apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx

```diff
@@ -0,0 +1,173 @@
+                                {emptyWalletDescription}
+                            </div>
+                        ) : null}
+                    </div>
+                </>
+            ) : (
+                <div className={st.emptyWallet}>
+                    <div className={st.emptyWalletIcon} />
+                    <div className={st.emptyWalletTitle}>
+                        Your wallet contains no SUI.
+                    </div>
+                    {emptyWalletDescription}
+                    <FaucetRequestButton />
+                </div>
+            )}
+            <FaucetMessageInfo className={st.gasRequestInfo} />
+        </Loading>
+    );
+}
+
+function TokenDetails({ coinType }: TokenDetailsProps) {
+    const { loading, error, showError } = useObjectsState();
+    const activeCoinType = coinType || GAS_TYPE_ARG;
+    const balances = useAppSelector(accountAggregateBalancesSelector);
+    const tokenBalance = balances[activeCoinType] || BigInt(0);
+    const allCoinTypes = useMemo(() => Object.keys(balances), [balances]);
+    const coinTypeWithBalance =
+        tokenBalance > 0 ? activeCoinType : allCoinTypes[0];
+
+    const coinSymbol = useMemo(
+        () => (coinType ? Coin.getCoinSymbol(coinType) : ''),
+        [coinType]
+    );
+    return (
+        <div className={st.container}>
+            {showError && error ? (
+                <Alert className={st.alert}>
+                    <strong>Sync error (data might be outdated).</strong>{' '}
+                    <small>{error.message}</small>
+                </Alert>
+            ) : null}
+            {!coinType && <AccountAddress showLink={false} mode="faded" />}
+            <div className={st.balanceContainer}>
+                <Loading loading={loading}>
+                    <CoinBalance
+                        balance={tokenBalance}
+                        type={activeCoinType}
+                        mode="standalone"
+                    />
+                </Loading>
+            </div>
+            <div className={st.actions}>
+                <IconLink
+                    icon={SuiIcons.Buy}
+                    to="/"
+                    disabled={true}
+                    text="Buy"
+                />
+                <IconLink
+                    icon={SuiIcons.ArrowLeft}
+                    to={`/send${
+                        coinTypeWithBalance
+                            ? `?${new URLSearchParams({
+                                  type: coinTypeWithBalance,
+                              }).toString()}`
+                            : ''
+                    }`}
+                    disabled={!coinTypeWithBalance}
+                    text="Send"
+                />
+                <IconLink
+                    icon={SuiIcons.Swap}
+                    to="/"
+                    disabled={true}
+                    text="Swap"
+                />
+            </div>
+
+            {coinType === GAS_TYPE_ARG || !coinType ? (
```

Comment:
```suggestion
            {activeCoinType === GAS_TYPE_ARG ? (
```
should be the same right?